source/common/common/utility.cc

```diff
@@ -123,6 +124,16 @@ std::vector<std::string> StringUtil::split(const std::string& source, const std:
   return ret;
 }
 
+// TODO: templatize this, no reason it needs to always be strings
```

Comment:
nit: I would remove TODO. This can't be trivially templatized due to use of ostringstream. We can deal with it later if the need arises.